examples/vm_scheduling/rule_based_algorithm/algorithm.py

```diff
@@ -0,0 +1,28 @@
+import math
+
+from maro.simulator import Env
+from maro.simulator.scenarios.vm_scheduling.common import Action
+from maro.simulator.scenarios.vm_scheduling import AllocateAction, DecisionPayload, PostponeAction
+
+
+class VMSchedulingAgent(object):
```

Comment:
the filename is `algorithm`, but the class is `agent`.

- extract an agent class, which maintains an algorithm instance.
- create an abstract algorithm class, which has an abstract `allocate_vm` function